src/main/java/apoc/cypher/Cypher.java

```diff
@@ -163,7 +203,7 @@ private Object executeStatement(BlockingQueue<RowResult> queue, String stmt, Map
                 queue.put(new RowResult(row++, result.next()));
             }
             if (addStatistics) {
-                queue.offer(new RowResult(-1, toMap(result.getQueryStatistics(), System.currentTimeMillis() - time, row)), timeout,TimeUnit.SECONDS);
+                queue.put(new RowResult(-1, toMap(result.getQueryStatistics(), System.currentTimeMillis() - time, row)));
```

Comment:
this is now blocking forever if the queue is full? 
exactly what we wanted to avoid with the timeout?
that was the reason we used offer in the first place ...

Reply:
Did you want to wrap this one too?